### DISCUSSION TITLE
Storybook 8: Using next/font/google when rendering storybook preview causes TypeError Cannot read properties of undefined (reading 'className')

### DISCUSSION THREAD

**Discussion by colmukrainec** [storybookjs/storybook#26462] [Help]
### Summary

The `next/font/google` imported font is causing `TypeError Cannot read properties of undefined (reading 'className')` on Storybook. I have searched and tried everything I could find including:

 - Reinstalling Storybook
 - Reproducing in a new project
 - Using `.variable` instead of `.className`
 - Upgrading to Storybook 8
 - Overriding `enhanced-resolve` to version 5.10 from 5.11
 - Importing the font locally
 - Importing the font from `next/fonts/google` directly in `preview.tsx`

File structure:

```
├── .storybook
│   ├── main.ts
│   ├── preview.tsx
├── src
│   ├── app
│   │   ├── layout.tsx
│   │   ├── page.tsx
│   │   ├── globals.css
│   ├── lib
│   │   ├── fonts.ts
│   ├── components
│   │   ├── Button
│   │   │   ├── Button.tsx
│   │   │   ├── Button.css
│   │   │   ├── Button.stories.ts
├── package.json
├── package-lock.json
├── tsconfig.json
└── tailwind.config.js
```

Note: Additionally I tried the default `stories` folder and files that are made automatically by running `npx storybook@latest init`

To reproduce:
1. Run `npx create-next-app@latest`
    - Choose Yes for every question expect the custom aliasing (the last prompt)
2. Run `npx storybook@latest init`
3. Make directory and file `./src/lib/font.ts`
4. Add the following code to `./src/lib/font.ts`:
```
import { Quando } from 'next/font/google'

export const quando = Quando({
  weight: '400',
  subsets: ['latin'],
  variable: '--font-quando',
})
```
5. The `src/app/layout.tsx` should be edited to appear as:
```
import type { Metadata } from "next";
import "./globals.css";
import { quando } from "@/lib/fonts";

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={quando.className}>{children}</body>
    </html>
  );
}
```
6. Run `npm run dev` the next web app
    - Should see the default next page with the Quando font properly loaded
7. The `.storybook/preview.ts` file should be edited to appear as:
```
import type { Preview } from "@storybook/react";
import { quando } from "../src/lib/fonts";

const preview: Preview = {
  parameters: {
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
  },
  decorators: [
    Story => (
      <main className={quando.className}>
        <Story />
      </main>
    ),
  ],
};

export default preview;
```
8. Rename `preview.ts` to `preview.tsx` to fix the errors
9. Terminate the storybook process and run `npm run storybook`
10. Get an error on every component that appears as:
```
TypeError
Cannot read properties of undefined (reading 'className')
Call Stack
 preview.decorators
  main.iframe.bundle.js:51:79
 hookified
  runtime.js:84:10607
 undefined
  runtime.js:100:3256
 undefined
  runtime.js:100:3717
 withOutline
  vendors-node_modules_pmmmwh_react-refresh-webpack-plugin_client_ErrorOverlayEntry_js-node_mod-bc4dd4.iframe.bundle.js:2986:620
 hookified
  runtime.js:84:10607
 undefined
  runtime.js:100:3256
 undefined
  runtime.js:100:3717
 withMeasure
  vendors-node_modules_pmmmwh_react-refresh-webpack-plugin_client_ErrorOverlayEntry_js-node_mod-bc4dd4.iframe.bundle.js:2563:14020
 hookified
  runtime.js:84:10607
```

### Additional information

main.ts:

```
import type { StorybookConfig } from "@storybook/nextjs";

const config: StorybookConfig = {
  stories: ["../src/**/*.mdx", "../src/**/*.stories.@(js|jsx|mjs|ts|tsx)"],
  addons: [
    "@storybook/addon-onboarding",
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@chromatic-com/storybook",
    "@storybook/addon-interactions",
  ],
  framework: {
    name: "@storybook/nextjs",
    options: {},
  },
  docs: {
    autodocs: "tag",
  },
  staticDirs: ["..\\public"],
};
export default config;
```

Note: This was untouched as it was autogenerated.

**Comment by valentinpalkovic** [comment]
Hi @colmukrainec 

My wild guess is that you are on a Windows machine. I can't reproduce the error even after following the reproduction steps. 

Does the error also occur if Storybook runs in Babel mode? To try this out, please create a `.babelrc` in the root of your project with the following content:

```
{
  "presets": ["next/babel"]
}
```

Next week, I have a Windows machine available, and then I will give it another try to reproduce the issue.

**Reply by davidlhw** [reply]
Thanks @valentinpalkovic.

This bug is reproducible on a Windows machine for me and your suggestion solves it.

Any idea what might be the cause of this weird issue?

**Reply by valentinpalkovic** [reply]
It seems we have a bug when Storybook runs in SWC mode. That was the reason to ask for a reproduction when running in Babel mode. Could the downvoters of my answer please respond, what the downvote should mean? Does it mean you don't want to run in Babel mode (that was also not the intention of the message. I just wanted to eliminate a theory) or wasn't it reproducible in Babel mode?

**Reply by davidlhw** [reply]
Thanks for the reply.

It seems like this solution only addresses the symptom of the issue with SWC mode and not the underlying root cause.

FWIW, my build failed after configuring to run in this custom babel mode. Seems like `next/font` has [conflicts with custom babel configs](https://nextjs.org/docs/messages/babel-font-loader-conflict).

**Reply by valentinpalkovic** [reply]
That's correct. I try to take a look of fixing the issue in SWC mode next week!

**Comment by valentinpalkovic** [comment]
I have identified the root cause and I am working on a fix. Please continue the discussion in the following issue if necessary: https://github.com/storybookjs/storybook/issues/26699
